cli/popper/commands/cmd_run.py

```diff
@@ -120,124 +112,148 @@
     default=popper.scm.get_git_root_folder()
 )
 @pass_context
-def cli(ctx, action, wfile, skip_clone, skip_pull, skip, reuse,
-        recursive, quiet, debug, dry_run, parallel, log_file,
-        with_dependencies, on_failure, runtime, workspace):
-    """Execute github actions workflows.
+def cli(ctx, **kwargs):
+    """Executes one or more pipelines and reports on their status.
     """
-    popper.scm.get_git_root_folder()
+    if os.environ.get('CI') == 'true':
+        # When CI is set,
+        log.info('Running in CI environment...')
+        occurences = parse_commit_message()
+        if occurences:
+            for params in get_params(occurences):
+                kwargs.update(params)
+                if kwargs['recursive']:
+                    log.warn('When CI is set, --recursive is ignored.')
+                    kwargs['recursive'] = False
+                prepare_pipeline(**kwargs)
+        else:
+            # If no special keyword is found, we run all the workflows,
+            # recursively.
+            kwargs['recursive'] = True
+            prepare_pipeline(**kwargs)
+    else:
+        # When CI is not set,
+        prepare_pipeline(**kwargs)
+
+
+def prepare_pipeline(**kwargs):
+    """Set parameters for the workflow execution.
+    """
+
+    def is_workflow(argument):
```

Comment:
would it be possible to extend this logic so that we also accept a folder name as an argument, so that we can do:

```
popper run workflows/mywf
```

and if there's a `main.workflow` or `.github/main.workflow` file in that folder, then we run it?

Reply:
@ivotron In the case of `github-actions-demo` workflow, if we do `popper run github-actions-demo`, the workflow fails, since we are outside of project root. I guess this is not good. Also .github/main.workflow is present in the root always. So what should be done?

Reply:
oh, I see. In that case then I think it's best to not support passing a folder name

Reply:
Yeah.